pyterrier/text.py

```diff
@@ -247,6 +247,67 @@ def slidingWindow(sequence : list, winSize : int, step : int) -> list:
     """
     return [x for x in list(more_itertools.windowed(sequence,n=winSize, step=step)) if x[-1] is not None]
 
+def qbs(
+        text_scorer_pipe : TransformerBase, 
+        text_attr : str = "text", 
+        summary_attr : str = "summary", 
+        num_psgs : int = 5, 
+        joinstr : str ='...') -> TransformerBase:
+    """
+    Applies query-biased summarisation, by applying the specified text scoring pipeline.
+
+    Arguments:
+     - text_scorer_pipe(TransformerBase): the pipeline for scoring passages in response to the query. Normally this applies passaging.
+     - text_attr(str): what is the name of the attribute that contains the text of the document
+     - summary_attr(str): what is the name of the attribute that should contain the query-biased summary for that document
+     - num_psgs(int): how many passages to select for the summary of each document
+     - joinstr(str): how to join passages for a given document together
+
+    Example::
+
+        # retrieve documents with text
+        br = pt.BatchRetrieve(index, metadata=['docno', 'text'])
+
+        # use Tf as a passage scorer on sliding window passages 
+        psg_scorer = ( 
+            pt.text.sliding(text_attr='text', length=15, prepend_attr=None) 
+            >> pt.text.scorer(body_attr="text", wmodel='Tf', background_index=index)
+        )
+        
+        # use psg_scorer for performing query-biased summarisation on docs retrieved by br 
+        retr_pipe = br >> pt.text.qbs(psg_scorer)
+
+    """
+    import pyterrier as pt
+    tsp = (
+        pt.apply.query(lambda row: row['query'].replace('"', ""))
```

Comment:
What's this for? If it's to remove query language, something more general is probably needed. And I'm not really sure why you'd want to